Problem Building CUDA Support on Windows 10 (x64)

Building libnd4j with cuda support seems to work fine after reverting [Commit 16162c6039a32feda283b1f556cf4418d8f23c35 ](https://github.com/deeplearning4j/libnd4j/commit/16162c6039a32feda283b1f556cf4418d8f23c35). 

I then setup the necessary env variables using:
```
export VISUAL_STUDIO_HOME="/c/Program Files (x86)/Microsoft Visual Studio 12.0/VC/"
export LIBND4J_HOME="/c/users/dubs/documents/stuff/libndj4"
export INCLUDE="$VISUAL_STUDIO_HOME/include;$LIBND4J_HOME/include;$LIBND4J_HOME/blas;$INCLUDE" 
export LIB="$VISUAL_STUDIO_HOME/lib/amd64;$LIBND4J_HOME/blasbuild/blas;$LIB"
```

And try to compile nd4j using:
` mvn clean install -DskipTests -Dmaven.javadoc.skip=true`

Which results in this error:
```
   Creating library jniNd4jBlas.lib and object jniNd4jBlas.exp
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dtrsm(__int64 *,int,int,int,int,int,int,int,double,__int64,int,__int64,int)" (?dtrsm@Nd4jBlas@@QEAAXPEA_JHHHHHHHN_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::strsm(__int64 *,int,int,int,int,int,int,int,float,__int64,int,__int64,int)" (?strsm@Nd4jBlas@@QEAAXPEA_JHHHHHHHM_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dtrmm(__int64 *,int,int,int,int,int,int,int,double,__int64,int,__int64,int)" (?dtrmm@Nd4jBlas@@QEAAXPEA_JHHHHHHHN_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::strmm(__int64 *,int,int,int,int,int,int,int,float,__int64,int,__int64,int)" (?strmm@Nd4jBlas@@QEAAXPEA_JHHHHHHHM_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dsyr2k(__int64 *,int,int,int,int,int,double,__int64,int,__int64,int,double,__int64,int)" (?dsyr2k@Nd4jBlas@@QEAAXPEA_JHHHHHN_JH1HN1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::ssyr2k(__int64 *,int,int,int,int,int,float,__int64,int,__int64,int,float,__int64,int)" (?ssyr2k@Nd4jBlas@@QEAAXPEA_JHHHHHM_JH1HM1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dsyrk(__int64 *,int,int,int,int,int,double,__int64,int,double,__int64,int)" (?dsyrk@Nd4jBlas@@QEAAXPEA_JHHHHHN_JHN1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::ssyrk(__int64 *,int,int,int,int,int,float,__int64,int,float,__int64,int)" (?ssyrk@Nd4jBlas@@QEAAXPEA_JHHHHHM_JHM1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dsymm(__int64 *,int,int,int,int,int,double,__int64,int,__int64,int,double,__int64,int)" (?dsymm@Nd4jBlas@@QEAAXPEA_JHHHHHN_JH1HN1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::ssymm(__int64 *,int,int,int,int,int,float,__int64,int,__int64,int,float,__int64,int)" (?ssymm@Nd4jBlas@@QEAAXPEA_JHHHHHM_JH1HM1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dgemm(__int64 *,int,int,int,int,int,int,double,__int64,int,__int64,int,double,__int64,int)" (?dgemm@Nd4jBlas@@QEAAXPEA_JHHHHHHN_JH1HN1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::sgemm(__int64 *,int,int,int,int,int,int,float,__int64,int,__int64,int,float,__int64,int)" (?sgemm@Nd4jBlas@@QEAAXPEA_JHHHHHHM_JH1HM1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dspr2(__int64 *,int,int,int,double,__int64,int,__int64,int,__int64)" (?dspr2@Nd4jBlas@@QEAAXPEA_JHHHN_JH1H1@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::sspr2(__int64 *,int,int,int,float,__int64,int,__int64,int,__int64)" (?sspr2@Nd4jBlas@@QEAAXPEA_JHHHM_JH1H1@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dsyr2(__int64 *,int,int,int,double,__int64,int,__int64,int,__int64,int)" (?dsyr2@Nd4jBlas@@QEAAXPEA_JHHHN_JH1H1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::ssyr2(__int64 *,int,int,int,float,__int64,int,__int64,int,__int64,int)" (?ssyr2@Nd4jBlas@@QEAAXPEA_JHHHM_JH1H1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dspr(__int64 *,int,int,int,double,__int64,int,__int64)" (?dspr@Nd4jBlas@@QEAAXPEA_JHHHN_JH1@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::sspr(__int64 *,int,int,int,float,__int64,int,__int64)" (?sspr@Nd4jBlas@@QEAAXPEA_JHHHM_JH1@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dsyr(__int64 *,int,int,int,double,__int64,int,__int64,int)" (?dsyr@Nd4jBlas@@QEAAXPEA_JHHHN_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::ssyr(__int64 *,int,int,int,float,__int64,int,__int64,int)" (?ssyr@Nd4jBlas@@QEAAXPEA_JHHHM_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dger(__int64 *,int,int,int,double,__int64,int,__int64,int,__int64,int)" (?dger@Nd4jBlas@@QEAAXPEA_JHHHN_JH1H1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::sger(__int64 *,int,int,int,float,__int64,int,__int64,int,__int64,int)" (?sger@Nd4jBlas@@QEAAXPEA_JHHHM_JH1H1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dtpsv(__int64 *,int,int,int,int,int,__int64,__int64,int)" (?dtpsv@Nd4jBlas@@QEAAXPEA_JHHHHH_J1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::stpsv(__int64 *,int,int,int,int,int,__int64,__int64,int)" (?stpsv@Nd4jBlas@@QEAAXPEA_JHHHHH_J1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dtbsv(__int64 *,int,int,int,int,int,int,__int64,int,__int64,int)" (?dtbsv@Nd4jBlas@@QEAAXPEA_JHHHHHH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::stbsv(__int64 *,int,int,int,int,int,int,__int64,int,__int64,int)" (?stbsv@Nd4jBlas@@QEAAXPEA_JHHHHHH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dtrsv(__int64 *,int,int,int,int,int,__int64,int,__int64,int)" (?dtrsv@Nd4jBlas@@QEAAXPEA_JHHHHH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::strsv(__int64 *,int,int,int,int,int,__int64,int,__int64,int)" (?strsv@Nd4jBlas@@QEAAXPEA_JHHHHH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dtpmv(__int64 *,int,int,int,int,int,__int64,__int64,int)" (?dtpmv@Nd4jBlas@@QEAAXPEA_JHHHHH_J1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::stpmv(__int64 *,int,int,int,int,int,__int64,__int64,int)" (?stpmv@Nd4jBlas@@QEAAXPEA_JHHHHH_J1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dtbmv(__int64 *,int,int,int,int,int,int,__int64,int,__int64,int)" (?dtbmv@Nd4jBlas@@QEAAXPEA_JHHHHHH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::stbmv(__int64 *,int,int,int,int,int,int,__int64,int,__int64,int)" (?stbmv@Nd4jBlas@@QEAAXPEA_JHHHHHH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dtrmv(__int64 *,int,int,int,int,int,double,__int64,int,__int64,int)" (?dtrmv@Nd4jBlas@@QEAAXPEA_JHHHHHN_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::strmv(__int64 *,int,int,int,int,int,float,__int64,int,__int64,int)" (?strmv@Nd4jBlas@@QEAAXPEA_JHHHHHM_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dspmv(__int64 *,int,int,int,double,__int64,__int64,int,double,__int64,int)" (?dspmv@Nd4jBlas@@QEAAXPEA_JHHHN_J1HN1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::sspmv(__int64 *,int,int,int,float,__int64,__int64,int,float,__int64,int)" (?sspmv@Nd4jBlas@@QEAAXPEA_JHHHM_J1HM1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dsbmv(__int64 *,int,int,int,int,double,__int64,int,__int64,int,double,__int64,int)" (?dsbmv@Nd4jBlas@@QEAAXPEA_JHHHHN_JH1HN1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::ssbmv(__int64 *,int,int,int,int,float,__int64,int,__int64,int,float,__int64,int)" (?ssbmv@Nd4jBlas@@QEAAXPEA_JHHHHM_JH1HM1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dsymv(__int64 *,int,int,int,double,__int64,int,__int64,int,double,__int64,int)" (?dsymv@Nd4jBlas@@QEAAXPEA_JHHHN_JH1HN1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::ssymv(__int64 *,int,int,int,float,__int64,int,__int64,int,float,__int64,int)" (?ssymv@Nd4jBlas@@QEAAXPEA_JHHHM_JH1HM1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dgbmv(__int64 *,int,int,int,int,int,int,double,__int64,int,__int64,int,double,__int64,int)" (?dgbmv@Nd4jBlas@@QEAAXPEA_JHHHHHHN_JH1HN1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::sgbmv(__int64 *,int,int,int,int,int,int,float,__int64,int,__int64,int,float,__int64,int)" (?sgbmv@Nd4jBlas@@QEAAXPEA_JHHHHHHM_JH1HM1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dgemv(__int64 *,int,int,int,int,double,__int64,int,__int64,int,double,__int64,int)" (?dgemv@Nd4jBlas@@QEAAXPEA_JHHHHN_JH1HN1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::sgemv(__int64 *,int,int,int,int,float,__int64,int,__int64,int,float,__int64,int)" (?sgemv@Nd4jBlas@@QEAAXPEA_JHHHHM_JH1HM1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::daxpy(__int64 *,int,double,__int64,int,__int64,int)" (?daxpy@Nd4jBlas@@QEAAXPEA_JHN_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::saxpy(__int64 *,int,float,__int64,int,__int64,int)" (?saxpy@Nd4jBlas@@QEAAXPEA_JHM_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dcopy(__int64 *,int,__int64,int,__int64,int)" (?dcopy@Nd4jBlas@@QEAAXPEA_JH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::scopy(__int64 *,int,__int64,int,__int64,int)" (?scopy@Nd4jBlas@@QEAAXPEA_JH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dscal(__int64 *,int,double,__int64,int)" (?dscal@Nd4jBlas@@QEAAXPEA_JHN_JH@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::sscal(__int64 *,int,float,__int64,int)" (?sscal@Nd4jBlas@@QEAAXPEA_JHM_JH@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::dswap(__int64 *,int,__int64,int,__int64,int)" (?dswap@Nd4jBlas@@QEAAXPEA_JH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::sswap(__int64 *,int,__int64,int,__int64,int)" (?sswap@Nd4jBlas@@QEAAXPEA_JH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::drotm(__int64 *,int,__int64,int,__int64,int,__int64)" (?drotm@Nd4jBlas@@QEAAXPEA_JH_JH1H1@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::srotm(__int64 *,int,__int64,int,__int64,int,__int64)" (?srotm@Nd4jBlas@@QEAAXPEA_JH_JH1H1@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::drotmg(__int64 *,__int64,__int64)" (?drotmg@Nd4jBlas@@QEAAXPEA_J_J1@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::srotmg(__int64 *,__int64,__int64)" (?srotmg@Nd4jBlas@@QEAAXPEA_J_J1@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::drotg(__int64 *,__int64)" (?drotg@Nd4jBlas@@QEAAXPEA_J_J@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::srotg(__int64 *,__int64)" (?srotg@Nd4jBlas@@QEAAXPEA_J_J@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::drot(__int64 *,int,__int64,int,__int64,int,double,double)" (?drot@Nd4jBlas@@QEAAXPEA_JH_JH1HNN@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: void __cdecl Nd4jBlas::srot(__int64 *,int,__int64,int,__int64,int,float,float)" (?srot@Nd4jBlas@@QEAAXPEA_JH_JH1HMM@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: int __cdecl Nd4jBlas::idamax(__int64 *,int,__int64,int)" (?idamax@Nd4jBlas@@QEAAHPEA_JH_JH@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: int __cdecl Nd4jBlas::isamax(__int64 *,int,__int64,int)" (?isamax@Nd4jBlas@@QEAAHPEA_JH_JH@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: double __cdecl Nd4jBlas::dasum(__int64 *,int,__int64,int)" (?dasum@Nd4jBlas@@QEAANPEA_JH_JH@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: float __cdecl Nd4jBlas::sasum(__int64 *,int,__int64,int)" (?sasum@Nd4jBlas@@QEAAMPEA_JH_JH@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: double __cdecl Nd4jBlas::dnrm2(__int64 *,int,__int64,int)" (?dnrm2@Nd4jBlas@@QEAANPEA_JH_JH@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: float __cdecl Nd4jBlas::snrm2(__int64 *,int,__int64,int)" (?snrm2@Nd4jBlas@@QEAAMPEA_JH_JH@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: float __cdecl Nd4jBlas::sdot(__int64 *,int,__int64,int,__int64,int)" (?sdot@Nd4jBlas@@QEAAMPEA_JH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: double __cdecl Nd4jBlas::ddot(__int64 *,int,__int64,int,__int64,int)" (?ddot@Nd4jBlas@@QEAANPEA_JH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: double __cdecl Nd4jBlas::dsdot(__int64 *,int,__int64,int,__int64,int)" (?dsdot@Nd4jBlas@@QEAANPEA_JH_JH1H@Z)
jniNd4jBlas.obj : error LNK2001: unresolved external symbol "public: float __cdecl Nd4jBlas::sdsdot(__int64 *,int,float,__int64,int,__int64,int)" (?sdsdot@Nd4jBlas@@QEAAMPEA_JHM_JH1H@Z)
C:\Users\dubs\Documents\stuff\nd4j\nd4j-backends\nd4j-backend-impls\nd4j-native\target\classes\org\nd4j\nativeblas\windows-x86_64\jniNd4jBlas.dll : fatal error LNK1120: 70 unresolved externals
```

--- a/blas/NativeBlas.h
+++ b/blas/NativeBlas.h
@@ -5,11 +5,12 @@
 #ifndef NATIVEOPERATIONS_NATIVEBLAS_H
 #define NATIVEOPERATIONS_NATIVEBLAS_H
 
+#include <dll.h>
 #include <pointercast.h>
 
 
 
-class Nd4jBlas {
+class ND4J_EXPORT Nd4jBlas {
 public:
 
     /*

--- a/blas/NativeOps.h
+++ b/blas/NativeOps.h
@@ -24,11 +24,12 @@
 # endif
 #endif
 
+#include <dll.h>
 #include <pointercast.h>
 
 
 
-class NativeOps {
+class ND4J_EXPORT NativeOps {
 
 
 public:

--- a/blas/cuda/NativeOps.cu
+++ b/blas/cuda/NativeOps.cu
@@ -162,6 +162,10 @@ public:
     }
 };
 
+#ifdef _WIN32
+ScalarShapeInformation ScalarInfo<double>::shapeInfo;
+ScalarShapeInformation ScalarInfo<float>::shapeInfo;
+#endif
 
 /**
  *

--- a/include/dll.h
+++ b/include/dll.h
@@ -6,5 +6,8 @@
 #define NATIVEOPERATIONS_DLL_H
 #ifdef _WIN32
 //#include <windows.h>
+#  define ND4J_EXPORT __declspec(dllexport)
+#else
+#  define ND4J_EXPORT
 #endif
 #endif //NATIVEOPERATIONS_DLL_H

